### DISCUSSION TITLE
Is WasmEnableThreads should work with .NET8 and Avalonia 11.1 ?

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#16776] [Q&A] [answered]
Hi,
I see that there are references to WasmEnableThreads in avalonia source code.
I haven't retried in 11.1 so I created a new project and added <WasmEnableThreads>true</WasmEnableThreads>
Hovewer I got the following error:
`Uncaught Error Error: Assert failed: Expect to have one js-module-threads asset in resources`

Is Wasm multithreading supposed to already work in Avalonia? may be I forgot something? or may be it is not supported yet?
Thank you.

**Comment by maxkatz6** [comment, marked as answer]
It works in .NET 8, you need to make sure all your cache is clear, but I wouldn't recommend it at all.

WASM threading was completely redone in .NET 9, it would be better to start from there.

**Reply by dbriard** [reply]
Good to know, thx, and I will double check my caches.
